agent/consul/status_endpoint.go

```diff
@@ -18,7 +19,14 @@ func (s *Status) Ping(args struct{}, reply *struct{}) error {
 }
 
 // Leader is used to get the address of the leader
-func (s *Status) Leader(args struct{}, reply *string) error {
+func (s *Status) Leader(args *structs.DCSpecificRequest, reply *string) error {
```

Comment:
Would msgpack error when trying to decode `struct{}` to a `DCSpecificRequest` (or vice versa)? That could happen if there was a version mismatch. The same applies below in Peers().

Reply:
Good Q. I think the answer is no you'd just end up with a zero value DCSpecificRequest which is handled below but would be good to have a test to simulate that and verify it maybe?

Reply:
I am inadvertently already testing this. The TestStatusLeader and TestStatusPeers tests I left unaltered which are still passing a struct{}{} instead of a `DCSpecificRequest`. Its also going through msgpackrpc so it is actually being encoded/decoded and it still works.